plugins/api/discovery/src/main/java/org/apache/cloudstack/discovery/ApiDiscoveryServiceImpl.java

```diff
@@ -84,6 +88,9 @@ public boolean start() {
             Set<Class<?>> cmdClasses = new LinkedHashSet<Class<?>>();
             for (PluggableService service : _services) {
                 s_logger.debug(String.format("getting api commands of service: %s", service.getClass().getName()));
+                if (service.getClass().getSimpleName().equals("QuotaServiceImpl") && PluginAccessConfigs.QuotaPluginEnabled.value()) {
```

Comment:
There are 80+ pluggable services. This QuotaServiceImpl check is very specific and can impact performance. Should look at a generic way to fetch APIs for a specific service